### DISCUSSION TITLE
How to extend webpack rule in Nuxt Bridge?

### DISCUSSION THREAD

**Discussion by yoshitsugu** [nuxt/nuxt#16103] [Questions]
Sorry for beginner's question. But how to extend webpack rule in Nuxt Bridge?
We can extend webpack rule like following in Nuxt 2:

```nuxt.config.js
{ 
  build: {
    extend(config, ctx) {
      config.module.rules.push({
        test: /\.(ogg|mp3|wav|mpe?g)$/i,
        loader: 'file-loader',
        options: {
          name: '[path][name].[ext]'
        }
      });
    }
  }
}
```

Is there alternative way to do same thing in Nuxt Bridge?
I'm now trying to upgrade my Nuxt 2 project to Nuxt Bridge.
I fixed current `nuxt.config.js` to `nuxt.config.ts` but type cheking says there is no type like `extend`.

**Comment by qucode1** [comment]
Did you ever find a solution for this?

**Reply by yoshitsugu** [reply]
Not yet :(

**Comment by wtagain** [comment]
Same question.
@danielroe

**Comment by AnthonyRuelle** [comment]
Same question ...

**Comment by keyakko** [comment]
In my environment, this problem was solved by using the `hooks` property of defineNuxtConfig.
The code example in question, when converted, looks like this.
```typescript
defineNuxtConfig({
  hooks: {
    'webpack:config'(configs) {
      if (configs[0] && configs[0].module && configs[0].module.rules) {
        configs[0].module.rules.push({
          test: /\.(ogg|mp3|wav|mpe?g)$/i,
          loader: 'file-loader',
          options: {
            name: '[path][name].[ext]',
          },
        })
      }
    },
  },
})
```
